src/components/buttons/accessible-button/accessible-button.js

```diff
@@ -1,73 +1,74 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import classnames from 'classnames';
-import styles from './accessible-button.mod.css';
+import { css } from '@emotion/core';
+import vars from '../../../../materials/custom-properties';
 
-/*
-  Even though this button could be a functional component a class is used to
-  support keeping a reference on the button for components using it.
-  React does not support this for functional components:
-    https://github.com/facebook/react/issues/4936
- */
-export default class AccessibleButton extends React.PureComponent {
-  static displayName = 'AccessibleButton';
-  static propTypes = {
-    id: PropTypes.string,
-    type: PropTypes.oneOf(['submit', 'reset', 'button']),
-    label: PropTypes.string.isRequired,
-    children: PropTypes.node.isRequired,
-    // set to true or false to indicate a toggle button
-    isToggled: ({ isToggleButton, isToggled }, ...rest) => {
-      if (isToggleButton && isToggled === undefined)
-        return new Error(
-          `\`isToggled\` is a required prop if \`isToggleButton\` is \`true\` on AccessibleButton component`
-        );
-      return PropTypes.bool({ isToggled }, ...rest);
-    },
-    buttonRef: PropTypes.func,
-    isToggleButton: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    onClick: PropTypes.func,
-    className: PropTypes.string,
-    // allows setting custom attributes on the underlying button html element
-    buttonAttributes: PropTypes.object,
-  };
+const AccessibleButton = React.forwardRef((props, ref) => (
+  <button
+    id={props.id}
+    ref={ref}
+    type={props.type}
+    aria-label={props.label}
+    onClick={props.onClick}
+    css={[
+      css`
+        border: none;
+        background: none;
+        display: inline-block;
+        outline: 0;
+        padding: 0;
+        margin: 0;
+        white-space: nowrap;
+        cursor: pointer;
+        color: inherit;
+        font: inherit;
+        font-size: ${vars['--font-size-default']};
+        font-family: ${vars['--font-family-default']};
+      `,
+      props.isDisabled &&
+        css`
+          cursor: not-allowed;
+        `,
+    ]}
```

Comment:
So is there a rule of thumb for us the extract these things?

Reply:
You mean extract into a separate file?

Reply:
Yeah. I guess it's just gut feeling.

Reply:
I find it useful to do once you start having more than a couple of long `getXxxStyles` functions, at which point it only pollutes the component file. Otherwise, for "small" styles I think it's better to have it within the component, for readability.